README.adoc

```diff
@@ -0,0 +1,144 @@
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/beers/package-info.java[]
+----
+
+== Implement the `BarmanService`
+
+Now you can implement the `BarmanService`.
+
+For `giveMeARandomBeer()`:
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/beers/impl/BarmanServiceImpl.java[tags=giveMeARandomBeer]
+----
+<1> Send a request to the https://www.craftbeernamegenerator.com[Craft Beer Name Generator Service]
+<2> Fail the async method if the request to external service failed
+<3> Create a new `Beer`
+<4> Add the beer to the customer bill
+<5> Complete the async method with the generated beer
+
+For `getMyBill()`:
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/beers/impl/BarmanServiceImpl.java[tags=getMyBill]
+----
+
+For `settleMyBill()`:
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/beers/impl/BarmanServiceImpl.java[tags=payMyBill]
+----
+
+== Expose the service
+
+Now we can expose the service inside the `BarmanVerticle` with the `ServiceBinder` class:
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/BarmanVerticle.java[]
+----
+<1> Instantiate the `BarmanServiceImpl`
+<2> Instantiate the `ServiceBinder`
+<3> Configure the service address
+<4> Register the service on the event bus
+
+== Use the service
+
+Now we can use the service with the generated proxy. For demo purposes I'm going to use it inside another verticle called `DrunkVerticle`:
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/DrunkVerticle.java[]
+----
+<1> Instantiate the proxy
+<2> Let's try the first beer 🍺
+<3> Something went wrong during beer retrieval, fail the verticle start
+<4> Drinking the first one 🍻
+<5> Give me another one 🍺
+<6> Drinking the second one 🍻
+<7> Time to go home. Give me the bill
+<8> Pay the bill
+
+== Running the application
+
+To run the application deploy the `BarmanVerticle` and subsequently the `DrunkVerticle`
+[source,java,indent=0,role="collapsed"]
+----
+include::src/main/java/io/vertx/howtos/ebservice/BeersApplication.java[]
+----
+
+You can run the application from:
+
+. your IDE, by running the `main` method from the `BeersApplication` class, or
+. with Maven: `mvn compile exec:java`
+
+You should see something like this:
+----
+The barman is ready to serve you
+Generated a new Beer! Sourpuss (American Malt Liquor)
+My first beer is a Sourpuss (American Malt Liquor) and it costs 5
+Generated a new Beer! Amateur Mountain IPA (American IPA)
+My second beer is a Amateur Mountain IPA (American IPA) and it costs 2
+My bill with the bar is 7
+Removed debt of homer
+----
+
+== Summary
+
+This how-to explained to you:
+
+. How to design and implement an Event Bus Service using Vert.x Service Proxy
```

Comment:
`. how to (...), and` for each line is better since you have an open sentence above (`to you:`).